cmd/namespace/delete.go

```diff
@@ -76,3 +90,71 @@ func (nc *NamespaceCommand) ExecuteDeleteNamespace(ctx context.Context, namespac
 	}
 	return nil
 }
+
+func (nc *NamespaceCommand) watchDelete(ctx context.Context, namespace string) error {
+	waitCtx, ctxCancel := context.WithCancel(ctx)
+	defer ctxCancel()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+
+	signal.Notify(stop, os.Interrupt)
+	exit := make(chan error, 1)
+	defer close(exit)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		exit <- nc.waitForNamespaceDeleted(waitCtx, namespace)
+	}(&wg)
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		err := nc.okClient.Stream().DestroyAllLogs(waitCtx, namespace)
+		if err != nil {
+			oktetoLog.Warning("delete namespace logs cannot be streamed due to connectivity issues")
+			oktetoLog.Infof("delete namespace logs cannot be streamed due to connectivity issues: %v", err)
+		}
+	}(&wg)
+
+	select {
+	case <-stop:
+		ctxCancel()
+		oktetoLog.Infof("CTRL+C received, exit")
+		return oktetoErrors.ErrIntSig
+	case err := <-exit:
+		// wait until streaming logs have finished
+		wg.Wait()
+		return err
+	}
+}
+
+func (nc *NamespaceCommand) waitForNamespaceDeleted(ctx context.Context, namespace string) error {
+	timeout := 5 * time.Minute
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(timeout)
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("'%s' deploy didn't finish after %s", namespace, timeout.String())
+		case <-ticker.C:
+			ns, err := nc.k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+			if err != nil {
+				// when err is NotFound return without error, as the namespace is correctly deleted
+				return nil
+			}
+
+			status, ok := ns.Labels["space.okteto.com/status"]
+			if !ok {
+				return errors.New("namespace does not have label for status")
+			}
+			if status == "DeleteFailed" {
+				return errors.New("namespace destroy all failed")
```

Comment:
Which is the error displayed to the user in these scenarios?

Reply:
i've updated this, when error in status Delete Failed `var errFailedDeleteNamespace = errors.New("failed to delete namespace")` i use this one, i am thinking that the one on the labels should probably be same? if there is no label we can not delete...